File: src/AdventureWorks.Models/Features/Sales/SalesTerritoryModel.cs
using AdventureWorks.Models.Features.AddressManagement;

namespace AdventureWorks.Models.Features.Sales;

public sealed class SalesTerritoryModel
{
    public int Id { get; set; }

    public string Name { get; set; }

    public string Group { get; set; }

    public decimal SalesYtd { get; set; }

    public decimal SalesLastYear { get; set; }

    public decimal CostYtd { get; set; }

    public decimal CostLastYear { get; set; }

    public CountryRegionModel CountryRegion { get; set; }
}

File: apps/angular-web/libs/shared/data-access/src/lib/models/lookup/sales-territory.model.ts
import { CountryRegion } from './country-region.model';

export interface SalesTerritory {
  readonly id: number;
  readonly name: string;
  readonly group: string;
  readonly salesYtd: number;
  readonly salesLastYear: number;
  readonly costYtd: number;
  readonly costLastYear: number;
  readonly countryRegion: CountryRegion;
}
